diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4c1841ae1..e054ad3d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,7 @@ endif()
 if(CMAKE_C_COMPILER_ID MATCHES Intel)
   option_with_flags(ENABLE_XHOST "Enable processor-specific optimization" ON "-xHost" "-march=native" "/arch:AVX2")
 else()
-  option_with_flags(ENABLE_XHOST "Enable processor-specific optimization" ON  "-march=native" "-xHost" "/arch:AVX2")
+  option_with_flags(ENABLE_XHOST "Enable processor-specific optimization" ON  "-march=native" "-xHost")
 endif()
 
 option_with_print(DISABLE_VXC "Don't compile first derivative code" OFF)
@@ -434,6 +434,7 @@ set(raw_sources_list
   )
 
 set(src_prefix "src/")
+set(scratch_prefix "scratch/")
 string(REGEX REPLACE "([^;]+)" "${src_prefix}\\1" sources_list "${raw_sources_list}")
 
 set(raw_sources_list_f03
@@ -447,7 +448,11 @@ set(raw_sources_list_f90
 #   system to be found
 file(READ ${src_prefix}xc.h _src_contents)
 string(REPLACE "<xc_version.h>" "\"xc_version.h\"" _quoted_src "${_src_contents}")
-file(WRITE ${PROJECT_BINARY_DIR}/${src_prefix}xc.h "${_quoted_src}")
+file(WRITE ${PROJECT_BINARY_DIR}/${scratch_prefix}xc.h "${_quoted_src}")
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${scratch_prefix}xc.h ${src_prefix}xc.h
+  TIMEOUT 5
+  WORKING_DIRECTORY ${PROJECT_BINARY_DIR})
 
 # STATIC/SHARED on below governed by BUILD_SHARED_LIBS
 add_library(xc ${sources_list})
@@ -483,9 +488,14 @@ execute_process(
   TIMEOUT 5
   WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
 execute_process(
-  COMMAND ${PERL_EXECUTABLE} scripts/get_funcs.pl ${src_prefix} ${gen_dir}
+  COMMAND ${PERL_EXECUTABLE} scripts/get_funcs.pl ${src_prefix} ${PROJECT_BINARY_DIR}/${scratch_prefix}
   TIMEOUT 60
   WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
+set(gen_files funcs_gga.c funcs_hyb_gga.c funcs_hyb_lda.c funcs_hyb_mgga.c funcs_key.c funcs_lda.c funcs_mgga.c libxc_docs.txt libxc_inc.f90 xc_funcs.h xc_funcs_worker.h)
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E copy_if_different ${gen_files} ${gen_dir}
+  TIMEOUT 5
+  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/${scratch_prefix})
 
 add_executable(xc-info "${src_prefix}/xc-info.c")
 target_link_libraries(xc-info xc)
diff --git a/scripts/get_funcs.pl b/scripts/get_funcs.pl
index 1d5c7b64e..f6964a8f8 100755
--- a/scripts/get_funcs.pl
+++ b/scripts/get_funcs.pl
@@ -50,8 +50,8 @@ foreach $func (@funcs){
       $s3 .= sprintf "  %s %-30s = %3s  ! %s\n", "integer, parameter ::",
         $deflist_f{$key}, $key, $deflist_c{$key};
 
-      $s5 .= sprintf "  %s %-30s = %3s  ! %s\n", "integer(c_int), parameter, public ::",
-      $deflist_f{$key}, $key, $deflist_c{$key};
+      $s5 .= sprintf "\n! %s\n  %s %-30s = %3s\n", $deflist_c{$key}, "integer(c_int), parameter, public ::",
+      $deflist_f{$key}, $key;
     }
 
     $s1 .= "extern xc_func_info_type xc_func_info_$t;\n";
